Unity-Technologies/ScriptableRenderPipeline
# Corerp/add contextmenu dispatcher

### Purpose of this PR
There is conflict on contextual menu on **Camera** component > **Remove Component** between Built-in, HDRP and Universal version. This dispatcher allow to solve which render pipeline should handle the call.

This is pretty rough and only solve the conflict on **Remove Component** on the **Camera**.
It can be extended later to include more cases. Though, when the dispatcher dispatch, it cannot go back and call the built-in implementation as code path already branched. So it is required that the Dispatcher re-implement the built-in implementation for new case or that we move this to trunc at some point.

---
### Testing status

**Manual Tests**: 
- Other: Tested manually with several different RenderPipelineAsset in GraphuicSettings (Note that there is possible conflict when there is also RenderPipelineAsset in Quality that do not match)

For graphic QA: Please load a regular scene on master. And reference in the manifest both UP and HDRP (from Github reposiroty). You shouldn't see any message complaining about duplicate Remove Component context menu (but you will have complain about meta file conflict)

---
### Comments to reviewers
If you have any idea to not have to instantiate the  IRemoveAdditionalDataContextualMenu in order to call the right one, I can change it now or in a following PR.
